lib/store/index.js

```diff
@@ -31,9 +31,18 @@ export class Store {
   @observable editor = atom.workspace.getActiveTextEditor();
   @observable grammar: ?atom$Grammar;
   @observable configMapping: Map<string, ?mixed> = new Map();
+  globalMode: boolean = Boolean(atom.config.get("Hydrogen.globalMode"));
 
   @computed
   get kernel(): ?Kernel {
+    if (this.globalMode) {
+      if (!this.grammar) return null;
+      const currentScopeName = this.grammar.scopeName;
+      return this.runningKernels.filter(
+        k => k.grammar.scopeName === currentScopeName
+      )[0];
+    }
```

Comment:
Jasmine was failing when I originally used simple `k.grammar === this.grammar` this made it work.